### PR TITLE
First try on conversion of Singular.jl arrays to Singular lists

### PR DESCRIPTION
**Pull request by raulepure**
Added a first try on conversion of Singular.jl arrays to Singular lists in order to be able to call more Singular libraries in Singular.jl

### CODE REVIEW

File: caller.cpp [hunk 1]
```diff
@@ -374,6 +389,15 @@ void singular_define_caller(jlcxx::Module & Singular)
...
     Singular.method("jl_array_to_intmat", &jl_array_to_intmat);
     Singular.method("copy_string_to_void", &copy_string_to_void);
+    Singular.method("jl_sideal_array_to_void", [] (jl_value_t * args_val,
+                                                   jl_value_t * types_val,
+                                                   ring R) {
+        auto origin = currRing;
+        rChangeCurrRing(R);
+        lists l = jl_sideal_array_to_list_helper(args_val, types_val);
+        rChangeCurrRing(origin);
+        return (void *) L;
```
Comment (tthsqe12) [inline caller.cpp:399]: typo `return (void *) l;`

**Review by tthsqe12 — COMMENTED** [review]
Seems to work with https://github.com/oscar-system/Singular.jl/pull/358, but I objects to naming this function `jl_sideal_array_to_list_helper`. What does it have to do with ideals? Maybe just the name `jl_array_to_list_helper`?

File: caller.cpp (hunk 1 above)
Comment (raulepure) [reply caller.cpp:399]: The typo will be fixed, the name is still from the first trials with arrays of ideals. I will change it immediately.

**Comment by tthsqe12** [comment]
@thofma Is there anything wrong with this? Maybe get some new binaries with this one and #11?

**Pull request merged**